src/gui/FileDialog.cpp

```diff
@@ -42,11 +42,11 @@ QString FileDialog::getOpenFileName(QWidget* parent, const QString& caption, QSt
         if (parent) {
             parent->activateWindow();
         }
-
-        if (!result.isEmpty()) {
+        if (!result.isEmpty() && m_nextSaveLastDir) {
```

Comment:
@TheZ3ro maybe that could go in a `saveLastDir` function, since that code is repeated 4 times?